lldb/include/lldb/Target/DynamicLoader.h

```diff
@@ -329,6 +330,11 @@ class DynamicLoader : public PluginInterface {
   /// safe to call certain APIs or SPIs.
   virtual bool IsFullyInitialized() { return true; }
 
+  /// Return the `start` function \b symbol in the dynamic loader module.
+  virtual llvm::Expected<lldb_private::Symbol> GetStartSymbol() {
```

Comment:
Should this be an llvm::Expected?  It's only going to be defined for DynamicLoaderDarwin today.  If some generic code wanted to use this, where it may not be applicable or defined for another DynamicLoader plugin, it'll need to consume the UnimplementedError.  Given the optionality of it, a std::optional seems more natural to me maybe?

Reply:
In my understanding, every dynamic loader should have an equivalent to a start symbol, that's why I went with `Expected` instead of `optional`. I believe that long term, each dynamic loader plugin would have to implement this method for their own platform and only the base class would have the `UnimplementedError`.

Reply:
I would DEFINITELY not bank on this assumption.  The world of environments a debugger needs to support is vast.